`TestIntegrations/SessionRecordingModes/BestEffortMode/BeforeStartFailure` flakiness

## Failure

#### Link(s) to logs

https://github.com/gravitational/teleport/actions/runs/11561658914/job/32181183223

#### Relevant snippet

```
{"timestamp":"2024-10-28T20:14:01Z","level":"debug","caller":"srv/sess.go:2281","message":"Attempting to create session tracker.","component":"session:node","session_id":"383671f5-5430-44e1-8812-bc40935beef8"}
Session terminating due to node error.
{"timestamp":"2024-10-28T20:14:01Z","level":"info","caller":"events/emitter.go:287","message":"emitting audit event","event_type":"session.data","fields":{"addr.local":"127.0.0.1:42803","addr.remote":"127.0.0.1:55984","code":"T2006I","ei":2147483646,"event":"session.data","login":"ci","namespace":"default","private_key_policy":"none","rx":2894,"server_hostname":"eaff83fa95b3","server_id":"00000000-0000-0000-0000-000000000000","server_version":"17.0.0-dev","sid":"383671f5-5430-44e1-8812-bc40935beef8","time":"2024-10-28T20:14:01.114Z","trace.component":"audit","tx":3436,"uid":"c1a93d76-121a-41a9-b5d5-b00352614b79","user":"ci","user_kind":1}}
{"timestamp":"2024-10-28T20:14:01Z","level":"info","caller":"resumption/server_exchange.go:138","message":"resumable connection completed"}
{"timestamp":"2024-10-28T20:14:01Z","level":"debug","caller":"resumption/server_exchange.go:144","message":"handling new resumable connection","error":"read loop\n\tpeer signaled connection close\n\t\tuse of closed network connection"}
    integration_test.go:978: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:978
        	            				/__w/teleport/teleport/integration/integration_test.go:1048
        	Error:      	Received unexpected error:
        	            	ssh: could not start shell
        	Test:       	TestIntegrations/SessionRecordingModes/BestEffortMode/BeforeStartFailure
{"timestamp":"2024-10-28T20:14:01Z","level":"debug","caller":"resumption/client.go:292","message":"handling new resumable connection","error":"write loop\n\tconnection closed\n\t\tuse of closed network connection","trace_id":"46076973847f8d06933f09069[915](https://github.com/gravitational/teleport/actions/runs/11561658914/job/32181183223#step:6:916)8b39","span_id":"62f6a7d3083e3352"}
            --- FAIL: TestIntegrations/SessionRecordingModes/BestEffortMode/BeforeStartFailure (0.05s)
FAIL integration.TestIntegrations/SessionRecordingModes/BestEffortMode/BeforeStartFailure (0.05s)
```

--- a/integration/helpers/fixture.go
+++ b/integration/helpers/fixture.go
@@ -54,7 +54,6 @@ func NewFixture(t *testing.T) *Fixture {
 	fixture.Priv, fixture.Pub, err = testauthority.GenerateKeyPair()
 	require.NoError(t, err)
 
-	// Find AllocatePortsNum free listening ports to use.
 	fixture.Me, err = user.Current()
 	require.NoError(t, err)
 

--- a/integration/helpers/instance.go
+++ b/integration/helpers/instance.go
@@ -1525,6 +1525,7 @@ func (i *TeleInstance) NewUnauthenticatedClient(cfg ClientConfig) (tc *client.Te
 		Stdout:                        cfg.Stdout,
 		NonInteractive:                true,
 		DisableSSHResumption:          cfg.DisableSSHResumption,
+		AddKeysToAgent:                client.AddKeysToAgentNo,
 	}
 
 	// JumpHost turns on jump host mode

--- a/integration/integration_test.go
+++ b/integration/integration_test.go
@@ -889,15 +889,16 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 
 	// startSession starts an interactive session, users must terminate the
 	// session by typing "exit" in the terminal.
-	startSession := func(username string) (*Terminal, chan error) {
+	startSession := func(login, teleportUser string) (*Terminal, chan error) {
 		term := NewTerminal(250)
 		errCh := make(chan error)
 
 		go func() {
 			cl, err := teleport.NewClient(helpers.ClientConfig{
-				Login:   username,
-				Cluster: helpers.Site,
-				Host:    Host,
+				Login:        login,
+				TeleportUser: teleportUser,
+				Cluster:      helpers.Site,
+				Host:         Host,
 			})
 			if err != nil {
 				errCh <- trace.Wrap(err)
@@ -946,10 +947,15 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 	} {
 		t.Run(name, func(t *testing.T) {
 			// Setup user and session recording mode.
-			username := suite.Me.Username
-			role, err := types.NewRole("devs", types.RoleSpecV6{
+			login := suite.Me.Username
+			// Use unique Teleport user and role names per subtest to avoid
+			// cross-test cache/state reuse when role mappings are updated.
+			id := uuid.NewString()[:8]
+			teleportUser := fmt.Sprintf("%s-%s", strings.ToLower(name), id)
+			roleName := fmt.Sprintf("devs-%s", id)
+			role, err := types.NewRole(roleName, types.RoleSpecV6{
 				Allow: types.RoleConditions{
-					Logins:     []string{username},
+					Logins:     []string{login},
 					NodeLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
 				},
 				Options: types.RoleOptions{
@@ -959,7 +965,7 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 				},
 			})
 			require.NoError(t, err)
-			require.NoError(t, helpers.SetupUser(teleport.Process, username, []types.Role{role}))
+			require.NoError(t, helpers.SetupUser(teleport.Process, teleportUser, []types.Role{role}))
 
 			t.Run("BeforeStartFailure", func(t *testing.T) {
 				// Enable disk failure.
@@ -967,7 +973,7 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 				defer disableDiskFailure()
 
 				// Start session.
-				term, errCh := startSession(username)
+				term, errCh := startSession(login, teleportUser)
 				if test.expectSessionFailure {
 					waitSessionTermination(t, errCh, require.Error)
 					return
@@ -990,7 +996,7 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 
 			t.Run("MidSessionFailure", func(t *testing.T) {
 				// Start session.
-				term, errCh := startSession(username)
+				term, errCh := startSession(login, teleportUser)
 
 				// Guarantee the session started properly.
 				select {
@@ -1016,6 +1022,7 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 				term.Type("exit\n\r")
 				waitSessionTermination(t, errCh, require.NoError)
 			})
+
 		})
 	}
 }

